File: AsyncTaskWS/app/src/main/java/br/unicamp/asynctaskws/AlunoJsonParser.java
package br.unicamp.asynctaskws;

import com.google.gson.JsonObject;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class AlunoJsonParser {
    public static ArrayList<Aluno> parseDados(String content) {
        try {
            try {
                JSONArray jsonArray = new JSONArray(content);
                ArrayList<Aluno> alunoList = new ArrayList<>();

                for (int i = 0; i < jsonArray.length(); i++) {
                    JSONObject jsonObject = jsonArray.getJSONObject(i);
                    Aluno aluno = new Aluno();

                    aluno.setRA(jsonObject.getString("RA"));
                    aluno.setNome(jsonObject.getString("nome"));
                    aluno.setCorreio(jsonObject.getString("correio"));

                    alunoList.add(aluno);
                }

                return alunoList;
            } catch (JSONException e) {
                JSONObject jsonObject = new JSONObject(content);

                ArrayList<Aluno> alunoList = new ArrayList<>();

                Aluno aluno = new Aluno();
                aluno.setRA(jsonObject.getString("RA"));
                aluno.setNome(jsonObject.getString("nome"));
                aluno.setCorreio(jsonObject.getString("correio"));

                alunoList.add(aluno);
                return alunoList;
            }
        } catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: AsyncTaskWS/app/src/main/java/br/unicamp/asynctaskws/AlunoJsonParser.kt
package br.unicamp.asynctaskws

import com.google.gson.JsonObject

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

object AlunoJsonParser {
    fun parseDados(content: String): ArrayList<Aluno>? {
        try {
            try {
                val jsonArray = JSONArray(content)
                val alunoList = ArrayList<Aluno>()

                for (i in 0 until jsonArray.length()) {
                    val jsonObject = jsonArray.getJSONObject(i)
                    val aluno = Aluno()

                    aluno.ra = jsonObject.getString("RA")
                    aluno.nome = jsonObject.getString("nome")
                    aluno.correio = jsonObject.getString("correio")

                    alunoList.add(aluno)
                }

                return alunoList
            } catch (e: JSONException) {
                val jsonObject = JSONObject(content)

                val alunoList = ArrayList<Aluno>()

                val aluno = Aluno()
                aluno.ra = jsonObject.getString("RA")
                aluno.nome = jsonObject.getString("nome")
                aluno.correio = jsonObject.getString("correio")

                alunoList.add(aluno)
                return alunoList
            }

        } catch (e: JSONException) {
            e.printStackTrace()
            return null
        }

    }
}
